Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -13,35 +13,62 @@
 
 use Neos\Cqrs\Event\EventInterface;
 use Neos\Cqrs\Event\EventTransport;
-use Neos\Cqrs\Event\EventType;
+use Neos\Cqrs\Event\EventTypeService;
 use Neos\Cqrs\Message\MessageMetadata;
 use TYPO3\Flow\Annotations as Flow;
-use TYPO3\Flow\Utility\Arrays;
 
 /**
  * Base class for an aggregate root
  */
 abstract class AbstractAggregateRoot implements AggregateRootInterface
 {
     /**
+     * @var EventTypeService
+     * @Flow\Inject
+     */
+    protected $eventTypeService;
+
+    /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateIdentifier;
 
     /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateName;
 
     /**
      * @var EventTransport[]
-     * @Flow\Transient
      */
     protected $events = [];
 
     /**
+     * When calling self::recordThatMethod in the contructor
+     * the object is not initialized so event are queued
+     *
```

Comment:
Better:

"contains a list of events which have been recorded while the aggregate hasn't been initialized yet."